### PR TITLE
[inline_comments_data_collection] Add CL arguments for patch and diff length thresholds

### LINKED ISSUE #4466: [inline_comments_data_collection] Add patch and diff length threshold as a CL argument for integration tests
Currently, we specify `patch_threshold` and `diff_length_threshold` in `main()`, but once https://github.com/mozilla/bugbug/pull/4454 is merged, we should make it a command line argument to be able to limit the length of the dataset when conducting integration tests in `scripts/integration_test.sh`.

### PR DESCRIPTION
**Pull request by benjaminmah**
Resolves #4466.

Adds CL arguments for `patch-threshold` and `diff-length-threshold`, as well as an integration test.

### CODE REVIEW

File: scripts/inline_comments_data_collection.py [hunk 1]
```diff
@@ -183,11 +184,38 @@ def process_comments(patch_threshold, diff_length_threshold):
...
+        "--diff-length-threshold",
+        type=int,
+        default=None,
+        help="Limit the maximum allowed diff length. No limit if not specified.",
+    )
+
+    args = parser.parse_args()
+
+    patch_threshold = (
+        args.patch_threshold if args.patch_threshold is not None else float("inf")
```
Comment (suhaibmujahid) [inline scripts/inline_comments_data_collection.py:204]:
```suggestion
        args.patch_threshold or float("inf")
```

We could make `0` as invalid value.

File: scripts/inline_comments_data_collection.py [hunk 2]
```diff
@@ -183,11 +184,38 @@ def process_comments(patch_threshold, diff_length_threshold):
...
+    )
+    diff_length_threshold = (
+        args.diff_length_threshold
+        if args.diff_length_threshold is not None
+        else float("inf")
+    )
```
Comment (suhaibmujahid) [inline scripts/inline_comments_data_collection.py:210]: Same as https://github.com/mozilla/bugbug/pull/4470#discussion_r1747604774

File: scripts/inline_comments_data_collection.py [hunk 3]
```diff
@@ -183,11 +184,38 @@ def process_comments(patch_threshold, diff_length_threshold):
...
+    parser.add_argument(
+        "--patch-threshold",
+        type=int,
+        default=None,
+        help="Limit the number of patches to process. No limit if not specified.",
+    )
+    parser.add_argument(
+        "--diff-length-threshold",
+        type=int,
+        default=None,
```
Comment (suhaibmujahid) [inline scripts/inline_comments_data_collection.py:197]: We may need to specify a default value.

File: scripts/integration_test.sh [hunk 4]
```diff
@@ -22,6 +22,14 @@ ls -lh data
 # Removes it to ensure the commit retrieval work as expected
 rm data/commit*
 
+# Then generate a test dataset of fixed inline comments
+bugbug-fixed-comments --patch-threshold 150 --diff-length-threshold 1000
```
Comment (suhaibmujahid) [inline scripts/integration_test.sh:26]: We may rely on the default value from the CLI for the diff length.

File: scripts/inline_comments_data_collection.py [hunk 5]
```diff
@@ -183,11 +184,38 @@ def process_comments(patch_threshold, diff_length_threshold):
 
 
 def main():
+    parser = argparse.ArgumentParser(description="Process patch reviews.")
+    parser.add_argument(
+        "--patch-threshold",
```
[the remark this answers is not in the document]
Comment (benjaminmah) [reply scripts/inline_comments_data_collection.py]: Done here: e4392a5b9ebbf6a998819b1bbca1e9fadcfe5816

File: scripts/inline_comments_data_collection.py [hunk 6]
```diff
@@ -183,11 +184,38 @@ def process_comments(patch_threshold, diff_length_threshold):
...
+        type=int,
+        default=None,
+        help="Limit the number of patches to process. No limit if not specified.",
+    )
+    parser.add_argument(
+        "--diff-length-threshold",
```
[the remark this answers is not in the document]
Comment (benjaminmah) [reply scripts/inline_comments_data_collection.py]: Done here: e4392a5b9ebbf6a998819b1bbca1e9fadcfe5816

File: scripts/inline_comments_data_collection.py (hunk 1 above)
Comment (benjaminmah) [reply scripts/inline_comments_data_collection.py]: Done here: 0e2cc4fe3fd318040c2174981498609b08737517

File: scripts/inline_comments_data_collection.py (hunk 2 above)
Comment (benjaminmah) [reply scripts/inline_comments_data_collection.py]: Done here: 0e2cc4fe3fd318040c2174981498609b08737517

File: scripts/inline_comments_data_collection.py (hunk 3 above)
Comment (benjaminmah) [reply scripts/inline_comments_data_collection.py]: Done here: 0e2cc4fe3fd318040c2174981498609b08737517

File: scripts/integration_test.sh (hunk 4 above)
Comment (benjaminmah) [reply scripts/integration_test.sh]: Done here: 0e2cc4fe3fd318040c2174981498609b08737517

**Review by suhaibmujahid — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**